static/js/src/advantage/react/components/Subscriptions/ExpiryNotification/ExpiryNotification.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from "react";
+import { mount } from "enzyme";
+
+import ExpiryNotification, {
+  ExpiryNotificationSize,
+} from "./ExpiryNotification";
+import { userSubscriptionStatusesFactory } from "advantage/tests/factories/api";
+import { Notification } from "@canonical/react-components";
+
+describe("ExpiryNotification", () => {
+  it("can display an expiring notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expiring: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expiring-large']").exists()).toBe(true);
+  });
+
+  it("can display a large expired notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_expired: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(wrapper.find("[data-test='is_expired-large']").exists()).toBe(true);
+  });
+
+  it("can display a grace-period notification", () => {
+    const statuses = userSubscriptionStatusesFactory.build({
+      is_in_grace_period: true,
+    });
+    const wrapper = mount(
+      <ExpiryNotification
+        size={ExpiryNotificationSize.Large}
+        statuses={statuses}
+      />
+    );
+    expect(
+      wrapper.find("[data-test='is_in_grace_period-large']").exists()
```

Comment:
I think we'd get more value from this test if we were testing results that the user actually cares about instead (actual text that's displayed).